### PR TITLE
Set up test environment for API

### PR DESCRIPTION
**Pull request by KyunginNa**
[Set up test environment for API](https://www.pivotaltracker.com/story/show/176031700)
### User Story
```
As a development team,
In order to unit test our code
We would like to set up Rspec
```
## Changes proposed in this pull request:
* Configuring Rack-Cors
* Configuring Rspec, Should Matchers, and Factory Bot

## Packages/Gems added
* gem 'pry-rails'
* gem 'pry-byebug'
* gem 'rspec-rails'
* gem 'shoulda-matchers'
* gem 'factory_bot_rails'
* gem 'rack-cors'

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,8 +7,14 @@ gem 'rails', '~> 6.0.3', '>= 6.0.3.4'
 gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 4.1'
 gem 'bootsnap', '>= 1.4.2', require: false
+gem 'rack-cors', require: 'rack/cors'
 
 group :development, :test do
+  gem 'pry-rails'
+  gem 'pry-byebug'
```
Comment (emtalen) [inline Gemfile]: You don't need both so you can remove pry-byebug

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }
+
+RSpec.configure do |config|
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.use_transactional_fixtures = true
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
+  config.include FactoryBot::Syntax::Methods
+  config.include Shoulda::Matchers::ActiveRecord, type: :model
```
Comment (emtalen) [inline spec/rails_helper.rb:24]: please add the reponse JSON module

File: Gemfile (hunk 1 above)
Comment (KyunginNa) [reply Gemfile]: Deleted now!

File: spec/rails_helper.rb (hunk 2 above)
Comment (KyunginNa) [reply spec/rails_helper.rb:24]: Yes, created module and added it to rails_helper :)

**Review by emtalen — APPROVED** [review]

**Pull request merged**